### PR TITLE
Add remaining models

### PR DESCRIPTION
**Pull request by jrhorn424**
We need to discuss this code. I'll make comments in the code, but a few general notes:
- Try to write more descriptive commit messages. When I rebase and I see you've changed a file without knowing why, it's hard to know which version is the "correct" one.
- Try to make more, smaller commits. For example, if you're adding rolify and it require taking out a join table definition, that's fine, but not it. If the changes you've made are unrelated to adding/removing something (like rolify), make a separate commit.
- It's especially important to mention why migrations are deleted.

We can't automatically merge this branch into master since there are so many conflicts. That largely unavoidable, but remember:
- When you start your days work, rebase against `master` to pull in any changes that have been committed there by others (just me at the moment). Also rebase against any published feature branches, since these may have changed as well.
- Smaller commits mean fewer conflicts. "Commit early and often".

### CODE REVIEW

**Comment by jrhorn424** [comment]
Let's do a code review together. You look at my code, and I'll look at yours. This is the best time to ask questions, since we can mark the code directly with comments. When we're both ready, we'll merge.

**Comment by jrhorn424** [comment]
I've extensively rewritten the history. I'll show you the DAG for the code later today. This means your local copy is probably broken, so we'll need to fix that. I'm not sure if any of these patches are correct, since I had no clue what the intent was for each commit.

One thing I noticed when resolving these conflicts is that we need to be consistent with the naming of our join tables. I prefer the format `parents_childs_table`. For example, `experimenters_experiments_table`. It's less clear what we should do in the case of many-to-many relationships. What do you think? I find `users_sessions_table` slightly more preferable than `sessions_users_table`, but I don't know why.

**Comment by mlfreer** [comment]
Names of joint tables are written consistency with ruby convention, that subjects should be in the alphabetic order.
If it's necessary to change i can do it  

Понедельник,  7 октября 2013, 2:20 -07:00 от Jeffrey Horn notifications@github.com:

> I've extensively rewritten the history. I'll show you the DAG for the code later today. This means your local copy is probably broken, so we'll need to fix that. I'm not sure if any of these patches are correct, since I had no clue what the intent was for each commit.
> One thing I noticed when resolving these conflicts is that we need to be consistent with the naming of our join tables. I prefer the format parents_childs_table. For example, experimenters_experiments_table. It's less clear what we should do in the case of many-to-many relationships. What do you think? I find users_sessions_table slightly more preferable than sessions_users_table, but I don't know why.
> —
> Reply to this email directly or  view it on GitHub .

## 

Миша Фреер

**Comment by jrhorn424** [comment]
Don't worry about changing the join table names, as long as we're following a convention.

File: Gemfile [hunk 1]
```diff
@@ -18,6 +18,8 @@ gem 'authority'           #authorization
 
 gem 'inherited_resources'
 
+gem 'simple_form', '~> 3.0.0.rc'
```
Comment (jrhorn424) [inline Gemfile:21]: This should not be here.

File: app/models/category.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Category < ActiveRecord::Base
+  has_and_belongs_to_many :experiments, :join_table => :categories_experiments
```
Comment (jrhorn424) [inline app/models/category.rb:2]: Probably should have some validations here. And normalization, especially normalization. Tags should [slugs](http://stackoverflow.com/questions/4357007/what-does-slug-mean).

File: app/models/profile.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 class Profile < ActiveRecord::Base
-  belongs_to :user
+  belongs_to :user, inverse_of: :profile
   validates :user, presence: true
 
-  validates :first_name, :last_name, :gender, :age, :ethnicity,
+validates :first_name, :last_name, :gender, :age, :ethnicity,
```
Comment (jrhorn424) [inline app/models/profile.rb:5]: The two space indent was correct.

File: config/initializers/rolify.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Rolify.configure do |config|
+  # By default ORM adapter is ActiveRecord. uncomment to use mongoid
+  # config.use_mongoid
```
Comment (jrhorn424) [inline config/initializers/rolify.rb:3]: Don't need this. I don't see us using mongoid in the near future. I started with it and couldn't get it to behave. I don't know what I was thinking.

File: config/initializers/rolify.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  
+  # Dynamic shortcuts for User class (user.is_admin? like methods). Default is: false
+  # Enable this feature _after_ running rake db:migrate as it relies on the roles table
+  # config.use_dynamic_shortcuts
```
Comment (jrhorn424) [inline config/initializers/rolify.rb:7]: We'll be sure to enable this after merge.

File: db/schema.rb [hunk 6]
```diff
@@ -62,4 +134,25 @@
...
+  add_index "users_experimenters", ["user_id", "experimenter_id"], name: "index_users_experimenters_on_user_id_and_experimenter_id"
+
+  create_table "users_subjects", id: false, force: true do |t|
+    t.integer "user_id"
+    t.integer "subject_id"
+  end
+
+  add_index "users_subjects", ["user_id", "subject_id"], name: "index_users_subjects_on_user_id_and_subject_id"
+
 end
```
Comment (jrhorn424) [inline db/schema.rb:158]: Please pay particular attention to the schema when you're comparing your work to my rebase. Point out any errors I committed so we can fix them.

File: app/models/category.rb (hunk 2 above)
Comment (mlfreer) [inline app/models/category.rb:2]: Should we add normalizer here?

Comment (jrhorn424) [inline app/models/category.rb:2]: Yes, normalizers go in models, and I think we should sanitize categories as
slugs.

**Comment by jrhorn424** [comment]
OK, you can `pull --rebase` to get my last commit, make any changes requested in this discussion, and push the commits to this branch. The pull request will be automatically updated. Once that's done, we can merge.

**Pull request merged**